src/xrpld/app/tx/detail/MPTokenIssuanceCreate.cpp

```diff
@@ -0,0 +1,142 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2024 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <xrpld/app/tx/detail/MPTokenIssuanceCreate.h>
+#include <xrpld/ledger/View.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <xrpl/protocol/st.h>
+
+namespace ripple {
+
+NotTEC
+MPTokenIssuanceCreate::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureMPTokensV1))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    if (ctx.tx.getFlags() & tfMPTokenIssuanceCreateMask)
+        return temINVALID_FLAG;
+
+    if (auto const fee = ctx.tx[~sfTransferFee])
+    {
+        if (fee > maxTransferFee)
+            return temBAD_TRANSFER_FEE;
+
+        // If a non-zero TransferFee is set then the tfTransferable flag
+        // must also be set.
+        if (fee > 0u && !ctx.tx.isFlag(tfMPTCanTransfer))
+            return temMALFORMED;
+    }
+
+    if (auto const metadata = ctx.tx[~sfMPTokenMetadata])
+    {
+        if (metadata->length() == 0 ||
+            metadata->length() > maxMPTokenMetadataLength)
+            return temMALFORMED;
+    }
+
+    // Check if maximumAmount is within unsigned 63 bit range
+    if (auto const maxAmt = ctx.tx[~sfMaximumAmount])
+    {
+        if (maxAmt == 0)
+            return temMALFORMED;
+
+        if (maxAmt > maxMPTokenAmount)
```

Comment:
Looking at this PR, I had an idea: what if we removed this check and allowed the amount fields on `MPTokenIssuance` and `MPToken` to hold up to `2^64-1` instead?

I know that slightly changes what we agreed to with @shawnxie999 regarding MPT Amounts, but it occurred to me that our rationale to restrict MPT amounts related to `STAmount` and the use of Number in math operations. We can keep that restriction (i.e., in an `STAmount`, MPT amounts cannot exceed `2^63-1`). However, I don't think we'll ever do math with the overall MPT issuance amounts (whether `MaximumAmount`, `OutstandingAmount`, or even the total held by a single user in an `MPToken` via `MPTAmount`).

The benefit of this change is that an issuer could create an issuance with a much larger supply (~2x more) and in theory a token holder could hold the total supply (though that's a secondary benefit I guess).

Unless I'm missing something, the only drawback to this subtle change would be a very unlikely case where (1) an MPT issuance's max supply is greater-than `2^63-1` and (2) someone wants to create a transaction (payment, offer, etc) that includes the total supply. This of course would be precluded by the `STAmount` limit, and thus wouldn't ever be possible (and thus wouldn't break anything). 

What's more, I can't imagine this ever actually needing to happen. But on the off-chance someone _did_ need this to happen, then they could simply create two different transactions with amounts of `2^63-1` each and achieve the same outcome.

WDYT?

Reply:
Yes, agreed. I also spoke more with @gregtatcam and @shawnxie999, I feel good about being able to expand this in the future if demand warrants.

@shawnxie999 This comment can be resolved.